sim-lib/src/lib.rs

```diff
@@ -1116,30 +1150,40 @@ struct PaymentResultLogger {
 impl PaymentResultLogger {
     fn new() -> Self {
         PaymentResultLogger {
-            ..Default::default()
+            activity_results: HashMap::new(),
         }
     }
 
     fn report_result(&mut self, details: &Payment, result: &PaymentResult) {
+        let activity_name = details.activity_prefix();
+        let activity_result = self
+            .activity_results
+            .entry(activity_name.clone())
+            .or_default();
+
         match result.payment_outcome {
-            PaymentOutcome::Success => self.success_payment += 1,
-            _ => self.failed_payment += 1,
+            PaymentOutcome::Success => activity_result.success_payment += 1,
+            _ => activity_result.failed_payment += 1,
         }
 
-        self.total_sent += details.amount_msat;
+        activity_result.total_sent += details.amount_msat;
     }
 }
 
 impl Display for PaymentResultLogger {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        let total_payments = self.success_payment + self.failed_payment;
-        write!(
-            f,
-            "Processed {} payments sending {} msat total with {:.2}% success rate.",
-            total_payments,
-            self.total_sent,
-            (self.success_payment as f64 / total_payments as f64) * 100.0
-        )
+        for (activity_name, activity_result) in &self.activity_results {
+            let total_payments = activity_result.success_payment + activity_result.failed_payment;
+            writeln!(
+                f,
+                "{} Processed {} payments sending {} msat total with {:.2}% success rate.",
```

Comment:
The intent behind [this comment](https://github.com/bitcoin-dev-project/sim-ln/pull/170#discussion_r1578293858) was to remove the names from `PaymentResultLogger` completely - we don't want to print these results on a per-activity basis, it'll be too spammy where the whole point here is to give a very high level aggregate.

I believe we can also remove `activity_name` from `Payment` completely if we don't have this logging?